### PR TITLE
Goblin animation

### PR DESCRIPTION
**Pull request by phukaokub**
- Add Goblin Animation
- Add walk animation when move
- Refactor FieldTile position argument
- Refactor Entity update and render
- Refactor entity json file for animation
- Refactor and add Animation class in Util.py

### CODE REVIEW

File: requirements.txt
Comment (krittintrs) [inline requirements.txt:1]: Please review the required packages again. I thing we didn't use this much

File: src/battleSystem/FieldTile.py [hunk 1]
```diff
@@ -4,32 +4,29 @@
 class FieldTile:
     def __init__(self, index, position):
         self.index = index  # Index in the row
-        self.position = position  # (x, y) coordinates for rendering
+        self.position = pygame.Vector2(position)  # (x, y) coordinates for rendering
```
Comment (krittintrs) [inline src/battleSystem/FieldTile.py:7]: will it be better if we separate `position` field into `x` and `y` field?

File: src/states/SelectAttackState.py [hunk 2]
```diff
@@ -106,6 +106,7 @@ def update(self, dt, events):
                         if self.selectAttackTile>=0 and self.effect.maxRange>0:
                             if self.field[self.avilableAttackTile[self.selectAttackTile]].is_occupied():
                                 self.player.ChangeAnimation("multi_attack")
+                                self.enemy.ChangeAnimation("death")
```
Comment (krittintrs) [inline src/states/SelectAttackState.py:109]: I think "death" animation should be use when the enemy is actually dead. If we don't have the damage animation, maybe go with blinking in red instead. (might be future improvement)

File: requirements.txt
Comment (phukaokub) [reply requirements.txt:1]: I fixed the issue with `pipreqs`

File: src/states/SelectAttackState.py (hunk 2 above)
Comment (phukaokub) [reply src/states/SelectAttackState.py:109]: Maybe further improvement

**Pull request merged**